pyo3-ffi/src/cpython/pylifecycle.rs

```diff
@@ -44,5 +44,56 @@ extern "C" {
     // skipped _Py_LegacyLocaleDetected
     // skipped _Py_SetLocaleFromEnv
 
-    // skipped _Py_NewInterpreter
 }
+
+#[cfg(Py_3_12)]
+pub const PyInterpreterConfig_DEFAULT_GIL: c_int = 0;
+#[cfg(Py_3_12)]
+pub const PyInterpreterConfig_SHARED_GIL: c_int = 1;
+#[cfg(Py_3_12)]
+pub const PyInterpreterConfig_OWN_GIL: c_int = 2;
+
+#[cfg(Py_3_12)]
+#[repr(C)]
+pub struct PyInterpreterConfig {
+    use_main_obmalloc: c_int,
+    allow_fork: c_int,
+    allow_exec: c_int,
+    allow_threads: c_int,
+    allow_daemon_threads: c_int,
+    check_multi_interp_extensions: c_int,
+    gil: c_int,
```

Comment:
These fields need to be `pub` I think:

```suggestion
    pub use_main_obmalloc: c_int,
    pub allow_fork: c_int,
    pub allow_exec: c_int,
    pub allow_threads: c_int,
    pub allow_daemon_threads: c_int,
    pub check_multi_interp_extensions: c_int,
    pub gil: c_int,
```

Reply:
Oops